cmd/tools/grafana/grafana.go

```diff
@@ -196,69 +214,87 @@ func exportDashboards(opts *options) error {
 	return nil
 }
 
-func importDashboards(opts *options) error {
+// This is recursive function call.
+func importDashboards(opts *options, dir string, folderId int64) error {
 
 	var (
-		files              []os.FileInfo
-		request, dashboard map[string]interface{}
-		data               []byte
-		err                error
+		files         []os.FileInfo
+		importedFiles int
+		err           error
 	)
 
-	if files, err = ioutil.ReadDir(opts.dir); err != nil {
+	if files, err = ioutil.ReadDir(dir); err != nil {
 		// TODO check for not exist
 		return err
 	}
 
-	fmt.Printf("preparing to import %d dashboards...\n", len(files))
-
 	for _, f := range files {
-
-		if !strings.HasSuffix(f.Name(), ".json") {
-			//fmt.Printf("Skipping [%s]...\n", f.Name())
-			continue
+		// This is for 7 mode dashboards, for any new folder addition, this need to be followed
+		if f.IsDir() && (f.Name() == strings.ReplaceAll(grafana7modeFolderTitle, " ", "")) {
+			if err = importDashboards(opts, path.Join(dir, f.Name()), opts.folder[1].folderId); err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+		} else {
+			importFiles(f, dir, folderId, &importedFiles)
 		}
```

Comment:
may be just calling function with different argument based on if else would be easier than recursive call. This doesn't look good use of recursion to me.

Reply:
removed recursion from import